src/components/generics/date-range-picker.tsx

```diff
@@ -0,0 +1,98 @@
+import { Box } from "@mui/material";
+import { DatePicker } from "@mui/x-date-pickers";
+import { DateTime } from "luxon";
+import { useEffect, useState } from "react";
+import { getTimeDifferenceInDays } from "../../utils/time-utils";
+
+/**
+ * Component properties
+ */
+interface DateRangePickerProps {
+  dateTimeNow: DateTime;
+  setDates: Function;
+  initialStartDate?: DateTime;
+  initialEndDate?: DateTime;
+}
+/**
+ * Date range picker component
+ *
+ * @param props DateRangePickerProps
+ */
+const DateRangePicker = (props: DateRangePickerProps) => {
+  const { dateTimeNow, setDates, initialStartDate, initialEndDate } = props;
+  const [startDate, setStartDate] = useState<DateTime | null | undefined>(
+    initialStartDate ? initialStartDate : dateTimeNow
+  );
+  const [endDate, setEndDate] = useState<DateTime | null | undefined>(
+    initialEndDate ? initialEndDate : dateTimeNow
+  );
```

Comment:
```suggestion
const [dateRange, setDateRange] = useState({
    start: initialStartDate ? initialStartDate : dateTimeNow,
    end: initialStartDate ? initialStartDate : dateTimeNow
})
```

Reply:
This is a nice idea to pass one prop instead of two props to child components. Are the types supposed to be missing though?

Reply:
I left it untyped because Typescript might be able to infer the DateTime type from initialStartDate / dateTimeNow. If it does not work, I'd try making a custom interface.

Reply:
Also, for this change, changes are needed in other components as well.

Reply:
Not to worry though, it's not a big deal.

Reply:
Or then it is a big deal... Couldn't implement these changes correctly because it introduces new bugs I couldn't solve. I'll leave it as it is for now.